src/app/services/patient.service.ts

```diff
@@ -0,0 +1,233 @@
+              building = true;
+            const entries: BundleEntry[] = [];
+            const complete = () => {
+              // Do nothing if still sending requests
+              if (building) return;
+              if (remaining <= 0) {
+                // If all done, send our final event and tell the subscribers we're done
+                subscriber.next(new PatientDataEvent('complete', entries, entries.length, entries.length));
+                // Cache this
+                this.patientData = entries;
+                subscriber.complete();
+              }
+              // Otherwise do nothing and continue to wait
+            };
+            for (const type in resourceQueries) {
+              // Declare local loop variables here
+              // flag if we have a total
+              let haveTotal = false,
+                // local records loaded
+                localLoaded = 0;
+              // Increment types remaining for both completion .
+              remaining++;
+              overallTotalRemaining++;
+              this.fhirClient.getAllRecords(type, resourceQueries[type]).subscribe(
+                (next) => {
+                  const loadedEntries = next.elements.filter((record) => {
+                    // Check to make sure it's a bundle entry
+                    return 'fullUrl' in record && 'resource' in record;
+                  }) as BundleEntry[];
+                  // If we have a known total, set that
+                  if (next.total && !haveTotal) {
+                    haveTotal = true;
+                    overallTotal += next.total;
+                    overallTotalRemaining--;
+                  }
+                  localLoaded += next.elements.length;
+                  loaded += next.elements.length;
+                  entries.push(...loadedEntries);
+                  // Send the event
+                  subscriber.next(
+                    new PatientDataEvent(
+                      'progress',
+                      loadedEntries,
+                      overallTotalRemaining <= 0 ? overallTotal : undefined,
+                      loaded
+                    )
+                  );
+                },
+                (error) => {
+                  subscriber.error(error);
+                },
+                () => {
+                  if (!haveTotal) {
+                    // If we're done, we known the total now
+                    haveTotal = true;
+                    overallTotal += localLoaded;
+                    overallTotalRemaining--;
+                  }
+                  remaining--;
+                  complete();
+                }
+              );
+            }
+            // Indicate all queries are away and the complete event can fire
+            building = false;
+            // Run the complete handler to see if everything completed immediately
+            complete();
+          }
+        );
+    }));
+  }
+  // // Gathering resources for patient bundle
```

Comment:
Extraneous comment that can be removed?

Reply:
There are still a few "rough edges" with the updated CSS that have to do with replacing existing stuff with Angular Material widgets. The search page no longer being centered is one of those - reworking the header/footer to not overlap when too small broke that and I haven't changed that.

The last chunk of comments is indeed junk (it's the original code before it was refactored into the current version) and can be removed.